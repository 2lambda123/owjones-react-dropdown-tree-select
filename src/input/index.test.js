import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import Input from './index'

describe('<Input/>', () => {
  it('renders tags', () => {
    const tags = [{_id: 'i1', label: 'l1'}, {_id: 'i2', label: 'l2'}]
    const tree = renderer
      .create(<Input tags={tags} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders input when no tags are passed', () => {
    const tree = renderer
      .create(<Input />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders placeholder', () => {
    const placeholderText = 'select something'
    const tree = renderer
      .create(<Input placeholderText={placeholderText} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('raises onchange', () => {
    const onChange = jest.fn()
    const wrapper = shallow(<Input onInputChange={onChange} />)
    wrapper.find('input').simulate('change', {target: {value: 'hello'}, persist: jest.fn()})
    expect(onChange).toBeCalledWith('hello')
  })
})
