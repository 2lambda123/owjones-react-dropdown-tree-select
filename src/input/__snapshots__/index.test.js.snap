// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> renders input when no tags are passed 1`] = `
<ul
  className="tag-list"
>
  <li
    className="tag-item"
  >
    <input
      className="search"
      onBlur={undefined}
      onChange={[Function]}
      onFocus={undefined}
      placeholder="Choose..."
      type="text"
    />
  </li>
</ul>
`;

exports[`<Input/> renders placeholder 1`] = `
<ul
  className="tag-list"
>
  <li
    className="tag-item"
  >
    <input
      className="search"
      onBlur={undefined}
      onChange={[Function]}
      onFocus={undefined}
      placeholder="select something"
      type="text"
    />
  </li>
</ul>
`;

exports[`<Input/> renders tags 1`] = `
<ul
  className="tag-list"
>
  <li
    className="tag-item"
  >
    <span
      className="tag"
    >
      l1
      <button
        className="tag-remove"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </span>
  </li>
  <li
    className="tag-item"
  >
    <span
      className="tag"
    >
      l2
      <button
        className="tag-remove"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </span>
  </li>
  <li
    className="tag-item"
  >
    <input
      className="search"
      onBlur={undefined}
      onChange={[Function]}
      onFocus={undefined}
      placeholder="Choose..."
      type="text"
    />
  </li>
</ul>
`;
